data_steward/utils/auth.py

```diff
@@ -42,3 +63,95 @@ def delegated_credentials(credentials, scopes, subject=None):
             subject=subject)
 
     return updated_credentials
+
+
+def get_impersonation_credentials(target_principal,
+                                  target_scopes=None,
+                                  key_file=None):
+    """
+    Get useful impersonation credentials.
+
+    The target principal email address is required.  If a key_file path is not provided,
+    attempts will be made to determine your valid credentials from the environment.
+
+    Potential target scopes when using impersonation are:
+        'https://www.googleapis.com/auth/bigquery',
+        'https://www.googleapis.com/auth/devstorage.read_write',
+
+    :param target_principal:  email address to impersonate.  typically, this is an
+        email address associated with a google service account.
+    :param target_scopes: a list of scopes that the impersonating account needs
+        to perform requested actions.  if no list is provided, it defaults to a
+        limited set of functionality.
+    :param key_file: path to a json service account google key file.  If provided, this key
+        file will be used to create source credentials.  If missing and the
+        GOOGLE_APPLICATION_CREDENTIALS environment variable is set, that service
+        account key file will be used.  If this environment variable is unset, your
+        user account default application credentials will be used.  This usage
+        is preferable.
+
+    :returns: valid impersonated credentials
+    """
+    if not target_principal:
+        raise RuntimeError(
+            "Cannot impersonate.  No impersonation target specified.")
+
+    if not target_scopes:
+        LOGGER.warn(f"Using default scopes: {DEFAULT_SCOPES}")
+        target_scopes = DEFAULT_SCOPES
+
+    if not isinstance(target_scopes, list):
```

Comment:
I tried to set `target_scopes=google.cloud.storage.Client.SCOPE` ([code](https://github.com/googleapis/python-storage/blob/master/google/cloud/storage/client.py#L96-L100)) and it didn't work- maybe something like this instead?
```suggestion
    if not type(target_scopes) in (list, tuple):
```

Reply:
I'll look at this.  I want it to always work!  I was just looking at the scopes defined as strings.  Seems like we should be able to use scopes the way you are using them too.

Reply:
ok.  so this is just a tuple.  it's not an object.  I can get that working pretty quick.

Reply:
@mark-velez check it now to see if it works as you would hope/expect.

Reply:
Yes- that does it!